pkg/controller/servicediscovery/servicediscovery_controller.go

```diff
@@ -0,0 +1,205 @@
+
+const (
+	serviceDiscoveryImage = "lighthouse-agent"
+	appName               = "submariner-lighthouse-agent"
+)
+
+// Add creates a new ServiceDiscovery Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileServiceDiscovery{
+		client: mgr.GetClient(),
+		scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("servicediscovery-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource ServiceDiscovery
+	err = c.Watch(&source.Kind{Type: &submarinerv1alpha1.ServiceDiscovery{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to secondary resource DaemonSet and requeue the owner ServiceDiscovery
+	err = c.Watch(&source.Kind{Type: &appsv1.DaemonSet{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &submarinerv1alpha1.ServiceDiscovery{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileServiceDiscovery implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileServiceDiscovery{}
+
+// ReconcileServiceDiscovery reconciles a ServiceDiscovery object
+type ReconcileServiceDiscovery struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a ServiceDiscovery object and makes changes based on the state read
+// and what is in the ServiceDiscovery.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileServiceDiscovery) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling ServiceDiscovery")
+
+	// Fetch the ServiceDiscovery instance
+	instance := &submarinerv1alpha1.ServiceDiscovery{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			daemonSet := &appsv1.DaemonSet{}
```

Comment:
same as above. Should be a deployment with pod than a DaemonSet.

Reply:
Done.